cmd/sovereignnode/chainSimulator/tests/vm/issue_test.go

```diff
@@ -0,0 +1,213 @@
+			},
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+
+	defer cs.Close()
+
+	err = cs.GenerateBlocks(200)
+	require.Nil(t, err)
+
+	nodeHandler := cs.GetNodeHandler(core.SovereignChainShardId)
+	systemScAddress, _ := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Decode("erd1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6gq4hu")
+
+	wallet, err := cs.GenerateAndMintWalletAddress(core.SovereignChainShardId, initialMinting)
+	require.Nil(t, err)
+	nonce := uint64(0)
+
+	deployedContractAddress := chainSimulatorIntegrationTests.DeployContract(t, cs, wallet.Bytes, &nonce, systemScAddress, "", issueWasmPath)
+	deployedContractAddressBech32, err := nodeHandler.GetCoreComponents().AddressPubKeyConverter().Encode(deployedContractAddress)
+	require.Nil(t, err)
+
+	issueCost, _ := big.NewInt(0).SetString(issuePrice, 10)
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, deployedContractAddress, issueCost, "issue", uint64(60000000))
+
+	accountSC, _, err := nodeHandler.GetFacadeHandler().GetAccount(deployedContractAddressBech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.True(t, accountSC.Balance == "0")
+
+	esdts, err := nodeHandler.GetFacadeHandler().GetAllIssuedESDTs(core.FungibleESDT)
+	require.Nil(t, err)
+	require.NotNil(t, esdts)
+	require.True(t, len(esdts) == 1)
+
+	tokenIdentifier := esdts[0]
+	setRolesArgs := "setRoles@" + hex.EncodeToString([]byte(tokenIdentifier))
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, deployedContractAddress, big.NewInt(0), setRolesArgs, uint64(60000000))
+
+	esdtsRoles, _, err := nodeHandler.GetFacadeHandler().GetESDTsRoles(deployedContractAddressBech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotNil(t, esdtsRoles)
+	require.True(t, len(esdtsRoles[tokenIdentifier]) == 3)
+	require.Equal(t, core.ESDTRoleLocalMint, esdtsRoles[tokenIdentifier][0])
+	require.Equal(t, core.ESDTRoleLocalBurn, esdtsRoles[tokenIdentifier][1])
+	require.Equal(t, core.ESDTRoleTransfer, esdtsRoles[tokenIdentifier][2])
+
+	expectedMintedAmount, _ := big.NewInt(0).SetString("123000000000000000000", 10)
+	mintTxArgs := "mint" +
+		"@" + hex.EncodeToString([]byte(tokenIdentifier)) +
+		"@" + hex.EncodeToString(expectedMintedAmount.Bytes())
+	chainSimulatorIntegrationTests.SendTransaction(t, cs, wallet.Bytes, &nonce, deployedContractAddress, big.NewInt(0), mintTxArgs, uint64(20000000))
+
+	accountSC, _, err = nodeHandler.GetFacadeHandler().GetAccount(deployedContractAddressBech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.True(t, accountSC.Balance == "0")
+
+	esdtSC, _, err := nodeHandler.GetFacadeHandler().GetAllESDTTokens(deployedContractAddressBech32, coreAPI.AccountQueryOptions{})
+	require.Nil(t, err)
+	require.NotEmpty(t, esdtSC)
+	require.Equal(t, expectedMintedAmount, esdtSC[esdts[0]].Value)
+}
+
+func TestSmartContract_IssueToken_MainChain(t *testing.T) {
```

Comment:
This is literally duplicated as the one above.
Let's only use a func which receives the chain simulator interface and does the same test for both main chain and sovereign chain.
Also, the main chain test should not be in sovereign chain package.

Reply:
I moved this but I cannot really reuse much stuff because for main chain I need to wait some blocks for meta notarization in some places so I can get the issued tokens/roles and NodeHandler should be meta for some api calls

Reply:
Actually I deleted this from this PR. Maybe will move this into rc branch someday